### ISSUE TITLE
 Can't resolve 'child_process' with electron app + Angular + TS

### ISSUE THREAD

**Issue by metal3d** [kubernetes-client/javascript#218]
Hi,

I'm trying to create an application that embed Angular in Electron. 

My package.json file includes `"@kubernetes/client-node": "^0.8.1"` and `"electron": "^4.0.8",`

I'm building a service that prepare the client:
```
import { Injectable } from '@angular/core';
import { Core_v1Api, KubeConfig } from '@kubernetes/client-node';

@Injectable({
  providedIn: 'root'
})
export class K8sService {
  public k8sApi;

  constructor() { 
      const kc = new KubeConfig();
      kc.loadFromDefault()
      this.k8sApi = kc.makeApiClient(Core_v1Api);
  }
}
```

Then I try to build the application:
```
ng build --prod
```

The given error is:
```

ERROR in ./node_modules/shelljs/src/exec-child.js
Module not found: Error: Can't resolve 'child_process' in '/home/pafer/Projects/mainsail/node_modules/shelljs/src'
ERROR in ./node_modules/shelljs/src/exec.js
Module not found: Error: Can't resolve 'child_process' in '/home/pafer/Projects/mainsail/node_modules/shelljs/src'
ERROR in ./node_modules/aws-sign2/index.js
Module not found: Error: Can't resolve 'crypto' in '/home/pafer/Projects/mainsail/node_modules/aws-sign2'
ERROR in ./node_modules/aws4/aws4.js
Module not found: Error: Can't resolve 'crypto' in '/home/pafer/Projects/mainsail/node_modules/aws4'
ERROR in ./node_modules/ecc-jsbn/index.js
Module not found: Error: Can't resolve 'crypto' in '/home/pafer/Projects/mainsail/node_modules/ecc-jsbn'
ERROR in ./node_modules/http-signature/lib/signer.js
Module not found: Error: Can't resolve 'crypto' in '/home/pafer/Projects/mainsail/node_modules/http-signature/lib'
ERROR in ./node_modules/http-signature/lib/verify.js
...
...
```

There is no webpack configuration (and I have to say that I really don't know how and why some issues says to use one).

Without Angular, using only TS, it seems to work (I can list namespaces for example).

Is there anything to do to make it working with Angular ?

Keep in mind that I'm using Angular inside Electron app, not through a real web browser.

**Comment by brendandburns** [comment]
I don't really know anything about this issue, but I searched around a little bit.  I think adding:

```json
"browser": {
   "child_process": false,
   "crypto": false
}
```

to your `package.json` file may fix things.

Ref: https://stackoverflow.com/questions/54162297/module-not-found-error-cant-resolve-crypto

I'm closing this, since this is an electron-specific issue unrelated to this library.

**Comment by metal3d** [comment]
Hi,
Thanks for you answer.

I already tried and it gives no effect. I really don't understand why it works with TS in electron, but why in an Angular service or class the problem happens.

If I'm making an external ts script, then I use "tsc" to compile, no problem, but it's complex to attach the class in Angular components afterward.

I tried to change many tsconfig options, package.json options, and nothing worked.

IMHO, it is not an electron-specific problem, because I can use the librairie in electron. It's seems to be a compatibility problem (or option to find) between Angular and the librairie. I think that kubernetes-client librairie could give "integration documentation or FAQ" to help.

If someone want to take a look :)
